storage/jellyfish-merkle/src/jellyfish_merkle_test.rs

```diff
@@ -578,3 +580,269 @@ fn test_1000_versions() {
     let seed: &[_] = &[1, 2, 3, 4];
     many_versions_get_proof_and_verify_tree_root(seed, 1000);
 }
+
+proptest! {
+    #![proptest_config(ProptestConfig::with_cases(10))]
+
+    #[test]
+    fn test_get_range_proof(
+        (btree, n) in btree_map(any::<HashValue>(), any::<AccountStateBlob>(), 1..1000)
+            .prop_flat_map(|btree| {
+                let len = btree.len();
+                (Just(btree), 0..len)
+            })
+    ) {
+        let db = MockTreeStore::default();
+        let tree = JellyfishMerkleTree::new(&db);
+
+        for (i, (key, value)) in btree.iter().enumerate() {
+            let (_root_hash, batch) = tree
+                .put_blob_set(vec![(*key, value.clone())], i as Version)
+                .unwrap();
+            db.write_tree_update_batch(batch).unwrap();
+        }
+
+        let nth_key = *btree.keys().nth(n).unwrap();
+        let version = (btree.len() - 1) as Version;
+
+        let proof = tree.get_range_proof(nth_key, version).unwrap();
+        verify_range_proof(
+            tree.get_root_hash(version).unwrap(),
+            btree.into_iter().take(n + 1).collect(),
+            proof,
+        );
+    }
+}
+
+/// Checks if we can construct the expected root hash using the entries in the btree and the proof.
+fn verify_range_proof(
```

Comment:
Discussed offline.. we thought we could implement an even dumber but easier to understand version, for the purpose of testing.